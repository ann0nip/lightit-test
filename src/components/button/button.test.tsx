import { fireEvent, render, cleanup } from '@testing-library/react'
import Button from './button.component'

const labelText = 'Label Text'
const mockHandlerClick = jest.fn()

afterEach(cleanup);

describe('Button Component', () => {
    it('should take a snapshot', () => {
        const { asFragment } = render(<Button label={labelText} handleClick={mockHandlerClick} />)

        expect(asFragment()).toMatchSnapshot()
    })

    it('should render the label', () => {
        const { container } = render(<Button label={labelText} handleClick={mockHandlerClick} />)

        expect(container).toHaveTextContent(labelText)
    })

    it('should run the onClick event', () => {
        const { getByText } = render(<Button label={labelText} handleClick={mockHandlerClick} />)
        const button = getByText(labelText)

        fireEvent.click(button)

        expect(mockHandlerClick.mock.calls).toHaveLength(1)
    })
})