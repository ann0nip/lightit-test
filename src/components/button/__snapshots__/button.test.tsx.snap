// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-2 cursor-pointer border-[#979797]"
    data-name="Label Text"
    data-testid="button-btn"
  >
    <div
      class="flex h-12 w-28 items-center justify-center rounded-md bg-white transition focus:outline-0 focus:drop-shadow-none"
    >
      <span>
        Label Text
      </span>
    </div>
  </div>
</DocumentFragment>
`;
